pkg/stdcopy/stdcopy.go

```diff
@@ -155,11 +160,18 @@ func StdCopy(dstout, dsterr io.Writer, src io.Reader) (written int64, err error)
 			}
 		}
 
+		// we might have an error from the daemon mixed up in our multiplexed http
+		// stream. if we do, return it.
+		if out == nil {
+			return written, fmt.Errorf("got error from daemon: %v", string(buf[stdWriterPrefixLen:frameSize+stdWriterPrefixLen]))
```

Comment:
`error from daemon` instead of `got error from daemon`
Maybe be helpful though to specify an error type for this.